### PR TITLE
Exploring Github CI

### PR DESCRIPTION
**Pull request by Tam-Pham**
- [x] Add extras_require in setup.py for test deps 

- [x] Use extras_require to import test deps in tox.ini

- [x] move doctest to tox.ini

- [x] remove pytest from Github CI Tests (only test with tox)

- [x] add doctest to Github CI

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         #    python-version: 3.8
 
     steps:
-    - uses: actions/checkout@master
+    - uses: actions/checkout@v2
```
Comment (DominiqueMakowski) [inline .github/workflows/tests.yml:25]: my guess is that master is always up-to-date whereas V2 might get out-dated when V3 comes out

Comment (Tam-Pham) [reply .github/workflows/tests.yml]: yeahh, i was just trying to see if it makes any difference. Reverting back to master

Comment (hungpham2511) [reply .github/workflows/tests.yml]: master is the latest development and might contain bugs. So normally its preferable to use a released and numbered version like v2.

Comment (DominiqueMakowski) [reply .github/workflows/tests.yml]: ok, but that means from time to time we'll need to check if there is a V3 out? we can change to V2 for checkout and the other action in all of the workflows

**Pull request merged**